src/api/v4/client.js

```diff
@@ -15,8 +15,12 @@ function getValidationError (code) {
 /**
  * This is the entry point for a CARTO.js application.
  *
- * A CARTO client allows managing layers and dataviews. It also takes care
- * of the communication between a CARTO.js application and the services in CARTO.
+ * A CARTO client allows managing layers and dataviews. Once you add a layer or a dataview, the client 
+ * handles the communication against the server in order to process those data.
+ * 
+ * This means the client is an asynchronous object: When any change occurs, it will send a request to the server
+ * and act accordingly by updating the views or throwing an error when appropriate. This whole process is known as the `reload cycle`.
```

Comment:
I would try to avoid talking about the "reload cycle". It's a bit too internal for me. We can mention that some methods are async because the client needs to sync / communicate with the server.